File: MaratonaTreinamento/MaratonaTreinamento/Model/Exercise.cs
using System;
using Xamarin.Forms;

namespace MaratonaTreinamento.Model
{
    public class Exercise
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string SiteOrigem { get; set; }
        public int DifficultyLevel { get; set; }
        public string NecessaryKnowledge { get; set; }
        public string Description { get; set; }
        public string AuthorsComment { get; set; }

        public string GetColor
        {
            get
            {
                switch (DifficultyLevel)
                {
                    case 1:
                        return Color.Green.ToHex().ToString();
                        break;
                    case 2:
                        return Color.Blue.ToHex().ToString();
                        break;
                    case 3:
                        return Color.Purple.ToHex().ToString();
                        break;
                    case 4:
                        return Color.Orange.ToHex().ToString();
                        break;
                    case 5:
                        return Color.Red.ToHex().ToString();
                        break;
                }
                return Color.Yellow.ToHex().ToString();
            }
        }
        public string GetImagebackground
        {
            get
            {
                switch (DifficultyLevel)
                {
                    case 1:
                        return "beginnerLevel.jpg";
                        break;
                    case 2:
                        return "postBeginnerLevel.jpg";
                        break;
                    case 3:
                        return "intermediateLevel.jpg";
                        break;
                    case 4:
                        return "postIntermediateLevel.png";
                        break;
                    case 5:
                        return "hardLevel.png";
                        break;
                }
                return "";
            }
        }
    }
}

File: flutter_project/lib/1-base/models/exercise.dart
import 'package:flutter_project/2-app/helpers/paths.dart';
import 'package:flutter_project/2-app/views/stylePages/appColors.dart';
import 'package:uuid/uuid.dart';

class Exercise {
  //#region Attributes
  late String id;
  int? difficultyLevel;
  String? title;
  String? siteOrigem;
  String? necessaryKnowledge;
  String? description;
  String? authorsComment;
  late bool isFavorited;
  //#endregion

  //#region Constructor
  Exercise() {
    id = Uuid().v4();
    isFavorited = false;
  }
  //#endregion

  //#region Encapsulation
  get getColor {
    switch (difficultyLevel) {
      case 2:
        return AppColors().blueCardColor;
      case 3:
        return AppColors().purpleCardColor;
      case 4:
        return AppColors().orangeCardColor;
      case 5:
        return AppColors().redCardColor;
      default:
        return AppColors().greenCardColor;
    }
  }

  get getImagebackground {
    switch (difficultyLevel) {
      case 1:
        return "${Paths().imagesPath}beginnerLevel.jpg";
      case 2:
        return "${Paths().imagesPath}postBeginnerLevel.jpg";
      case 3:
        return "${Paths().imagesPath}intermediateLevel.jpg";
      case 4:
        return "${Paths().imagesPath}postIntermediateLevel.png";
      case 5:
        return "${Paths().imagesPath}hardLevel.png";
    }
  }

  get getStarColor {
    if (isFavorited)
      return AppColors().redCardColor;
    return AppColors().whiteColor;
  }
  //#endregion
}